docs/src/examples/jump_umap_analysis_with_cosmicqc.py

```diff
@@ -0,0 +1,421 @@
+    df=df_merged_single_cells,
+    metadata_columns=metadata_cols,
+    feature_thresholds="large_nuclei",
+)
+
+# label outliers within the dataset
+print("Elongated nuclei outliers:")
+df_labeled_outliers = cosmicqc.analyze.find_outliers(
+    df=df_merged_single_cells,
+    metadata_columns=metadata_cols,
+    feature_thresholds="elongated_nuclei",
+)
+
+# label outliers within the dataset
+print("Small and low formfactor nuclei outliers:")
+df_labeled_outliers = cosmicqc.analyze.find_outliers(
+    df=df_merged_single_cells,
+    metadata_columns=metadata_cols,
+    feature_thresholds="small_and_low_formfactor_nuclei",
+)
+
+# label outliers within the dataset
+df_labeled_outliers = cosmicqc.analyze.label_outliers(
+    df=df_merged_single_cells,
+    include_threshold_scores=True,
+)
+# show added columns
+df_labeled_outliers[
+    [col for col in df_labeled_outliers.columns.tolist() if "cqc." in col]
+].head()
+
+# +
+# create a column which indicates at least one outlier being found
+df_labeled_outliers["analysis.included_at_least_one_outlier"] = df_labeled_outliers[
+    [col for col in df_labeled_outliers.columns.tolist() if ".is_outlier" in col]
+].any(axis=1)
+
+# show value counts for all outliers
+outliers_counts = df_labeled_outliers[
+    "analysis.included_at_least_one_outlier"
+].value_counts()
+outliers_counts
+# -
+
+# show the percentage of total dataset
+print(
+    (outliers_counts.iloc[1] / outliers_counts.iloc[0]) * 100,
+    "%",
+    "of",
+    outliers_counts.iloc[0],
+    "include erroneous outliers of some kind.",
+)
+
+# show histograms to help visualize the data
+df_labeled_outliers.show_report()
+# +
+parquet_sampled_with_outliers = "./BR00117012_sampled_with_outliers.parquet"
+
+# check if we already have normalized data
+if not pathlib.Path(parquet_sampled_with_outliers).is_file():
+    # set a fraction for sampling
+    sample_fraction = 0.44
+
+    # read the dataset
+    df_features = pd.read_parquet("./BR00117012.parquet")
+
+    # group by metadata_well for all features then sample
+    # the dataset by a fraction.
+    df_features = (
+        df_features.groupby(["Metadata_Well"])[df_features.columns]
+        .apply(lambda x: x.sample(frac=sample_fraction))
+        .reset_index(drop=True)
+    )
+
+    # join the sampled feature data with the cosmicqc outlier data
+    df_feature_selected_with_cqc_outlier_data = df_features.merge(
+        df_labeled_outliers,
+        how="inner",
+        left_on=[
+            "Metadata_ImageNumber",
+            "Metadata_ObjectNumber",
+            "Metadata_Plate",
+            "Metadata_Well",
+        ],
+        right_on=[
+            "Metadata_ImageNumber",
+            "Metadata_ObjectNumber",
+            "Metadata_Plate",
+            "Metadata_Well",
+        ],
+    )
+
+    df_feature_selected_with_cqc_outlier_data.to_parquet(parquet_sampled_with_outliers)
+
+else:
+    df_feature_selected_with_cqc_outlier_data = pd.read_parquet(
```

Comment:
When you reran this to confirm the non-duck shape, did you start from scratch? (or did you use the existing parquet file)

Reply:
Thanks, I did re-run everything in full to confirm. I added the checks here to help speed up subsequent steps and as a safety from kernel crashing.